src/data_source/storage/sql.rs

```diff
@@ -1789,6 +1800,534 @@ impl SqlStorage {
+        let result = row_stream.map(|row| -> QueryResult<BlockSummary<Types>> {
+            let block = parse_block::<Types>(row?)?;
+            Ok(BlockSummary::try_from(block)?)
+        });
+
+        result.try_collect().await
+    }
+
+    async fn get_block_detail(
+        &self,
+        request: &BlockIdentifier<Types>,
+    ) -> QueryResult<BlockDetail<Types>> {
+        let (query, params): (String, Vec<Box<dyn ToSql + Sync + Send>>) = match request {
+            BlockIdentifier::Latest => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![],
+            ),
+            BlockIdentifier::Height(height) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        WHERE h.height = $1
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![Box::new(*height as i64)],
+            ),
+            BlockIdentifier::Hash(hash) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        WHERE h.hash = $1
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![Box::new(hash.to_string())],
+            ),
+        };
+
+        let query_result = self.query_one(&query, params).await?;
+        let block = parse_block(query_result)?;
+
+        Ok(BlockDetail::try_from(block).map_err(|e| QueryError::Error {
+            message: e.to_string(),
+        })?)
+    }
+
+    async fn get_transaction_summaries(
+        &self,
+        request: &GetTransactionSummariesRequest<Types>,
+    ) -> QueryResult<Vec<TransactionSummary<Types>>> {
+        let range = &request.range;
+        let target = &range.target;
+        let filter = &request.filter;
+
+        // We need to figure out the transaction target we are going to start
+        // returned results based on.
+        let transaction_target = match target {
+            TransactionIdentifier::Latest => self.query_one::<str, Vec<Box<dyn ToSql + Send + Sync>>>(
+                "SELECT t.block_height AS height, t.index AS index FROM transaction AS t ORDER BY (t.block_height, t.index) DESC LIMIT 1",
+                vec![],
+            ).await,
+            TransactionIdentifier::HeightAndOffset(height, offset) => self.query_one(
+                "SELECT t.block_height AS height, t.index AS index FROM transaction AS t WHERE t.block_height = $1 ORDER BY (t.block_height, t.index) DESC OFFSET $2 LIMIT 1",
```

Comment:
I think block_height column in transactions should have a separate  index

Reply:
It currently does not look to have a separate one.  It only has an index on the `hash`, and on the combination of `(block_height, index)`.

Unless I'm misunderstanding this comment and you're suggesting that we should add a new index for just `block_height`.

Reply:
yes we should add a separate btree index for blockheight

Reply:
In my local testing of the database for the sequencer, adding an index for this yields no speed up, and in fact seems to make things marginally slower.  I figure since there is already a compound index for `(block_height, index)` that the `block_height` is kind of already indexed in this situation.  Though my experience with using partially applied compound indexes tends to indicate that this isn't usually the case.

Sampling the performance by running this query:
```sql
SELECT t.block_height AS height, t.index AS index FROM hotshot.transaction AS t WHERE t.block_height = 20 ORDER BY (t.block_height, t.index) DESC OFFSET 0 LIMIT 1
```

If I run the query early (with a minimal amount of blocks saved to the database), I do get a more complex query plan:
<img width="517" alt="image" src="https://github.com/EspressoSystems/hotshot-query-service/assets/3937524/f88d3445-2664-4dba-a2f4-4a76493b3f25">

But this is likely fine, as there really isn't enough data to really speed up any execution via the query that a linear scan of fewer than 1500 indexes would still be fairly fast, taking at worst `83ms` on my hardware.

Once the demo stores above 1500 indexes the query plan changes:
<img width="598" alt="image" src="https://github.com/EspressoSystems/hotshot-query-service/assets/3937524/8aff9f77-00f0-4820-a1f2-db7f5441c357">

And looks fairly similar to the query plan from the added index.  It takes around `36.1ms` on average.

If add the index via:
```sql
CREATE INDEX transaction_block_height ON hotshot.transaction (block_height)
```

The query plan does in fact change:
<img width="620" alt="image" src="https://github.com/EspressoSystems/hotshot-query-service/assets/3937524/3c7c2425-b326-4d0a-87a0-3210e9bf852d">

Which takes about `43.8ms` on average to execute with the index added.

So we get slower queries, but for a "cheaper" query cost.

I'm not sure what to make of that.  What do you think @imabdulbasit?

Reply:
I think it is using the index because the first column in compound index is block height. I have had better read performance when adding separate index to table having millions of rows but that also slows down the upserts. I am keeping a note of this and going to investigate when we have a large db. Resolving this for now.